src/FiberTraces.cc

```diff
@@ -2683,62 +2683,49 @@ namespace pfsDRPStella = pfs::drp::stella;
+      size_t nRows = std::accumulate( thisFiberTrace.begin(), thisFiberTrace.end(), 0 );
       for (size_t i = 0; i < fiberTraceSet.size(); ++i) {
         PTR( FiberTrace<ImageT, MaskT, VarianceT> ) fiberTrace = fiberTraceSet.getFiberTrace( i );
         iTrace = findITrace( *fiberTrace,
                              xCenters,
-                             yCenters,
                              fiberTraceIds.getShape()[ 0 ],
                              nRows,
                              startPos );
         startPos = iTrace + 1;
-        cout << "FiberTraces::assignITrace: i = " << i << ": iTrace = " << iTrace << endl;
-        if ( iTrace < 0 ){
-          cout << "FiberTraces::assignITrace: i = " << i << ": trace not found => returning FALSE" << endl;
-          return false;
-        }
+        #ifdef __DEBUG_ASSIGNITRACE__
+          cout << "FiberTraces::assignITrace: i = " << i << ": iTrace = " << iTrace << endl;
+        #endif
         fiberTrace->setITrace( iTrace );
       }
-      return true;
+      return;
     }
 
     template< typename ImageT, typename MaskT, typename VarianceT, typename U, int I >
-    int findITrace( FiberTrace< ImageT, MaskT, VarianceT > const& fiberTrace,
-                    ndarray::Array< U, 1, I > const& xCentersArr,
-                    ndarray::Array< U, 1, I > const& yCentersArr,
-                    int nTraces,
-                    int nRows,
-                    int startPos ){
-      #ifdef __DEBUG_FINDITRACE__
-        cout << "FiberTraces::findITrace: fiberTraceIds = " << fiberTraceIds << endl;
-      #endif
-      float xCenter;
-      unsigned int yCenter;
-      xCenter = fiberTrace.getFiberTraceFunction()->xCenter;
-      yCenter = fiberTrace.getFiberTraceFunction()->yCenter;
+    size_t findITrace( FiberTrace< ImageT, MaskT, VarianceT > const& fiberTrace,
+                       ndarray::Array< U, 1, I > const& xCentersArr,
+                       size_t nTraces,
+                       size_t nRows,
+                       size_t startPos ){
+      float xCenter = fiberTrace.getFiberTraceFunction()->xCenter;
+      unsigned int yCenter = fiberTrace.getFiberTraceFunction()->yCenter;
       #ifdef __DEBUG_FINDITRACE__
         cout << "FiberTraces::findITrace: xCenter = " << xCenter << endl;
         cout << "FiberTraces::findITrace: yCenter = " << yCenter << endl;
       #endif
 
-      for ( int i = startPos; i < nTraces; ++i ){
-        if ( std::fabs( xCentersArr[ i * nRows + yCenter ] - xCenter ) < 2. ){
-          #ifdef __DEBUG_FINDITRACE__
-            cout << "FiberTraces::findITrace: i = " << i << ": returning i=" << i << endl;
-          #endif
-          return i;
-        }
+      ndarray::Array<float, 1, 1> dist = ndarray::allocate(nTraces - startPos - 1);
+      auto itDist = dist.begin();
+      for ( size_t i = startPos; i < nTraces; ++i, ++itDist ){
+        *itDist = std::fabs( xCentersArr[ i * nRows + yCenter ] - xCenter );
```

Comment:
This seems like a pretty memory intensive way to search. Wouldn't it be easier to use `std::min_element`? I assume you can get away with this here because `nTraces` is small, but it looks like a pattern to avoid in general.